### ISSUE TITLE
Use module loader

### ISSUE THREAD

**Issue by Ryuno-Ki** [zhukov/webogram#1300]
Since @zhukov said he would accept PR in that direction, I'd take a stab and suggest to introduce a module loader.
He asked in https://github.com/zhukov/webogram/issues/1292#issuecomment-268005299 to discuss about the pros and cons about several solution first.

Solutions could be:
* RequireJS
* Webpack
* Browserify
* Something else (Rollup? Svelte?)

What are the requirements?
* Runs in a browser/as chrome extension/as Firefox OS app/…?
* Takes care for module ordering and loading (Rich Harris coined the term [tree shaking](https://medium.com/@Rich_Harris/tree-shaking-versus-dead-code-elimination-d3765df85c80#.ab25q998h) for that IIRC)
* Works together with gulp
* Supports Angular 1
* …?

**Comment by Ryuno-Ki** [comment]
@marco-c wrote in https://github.com/zhukov/webogram/issues/1298#issuecomment-268038228
> Oh, I've noticed you're using the global scope there.
With service workers, it isn't reliable to use variables from the global scope in event handlers (see https://developer.mozilla.org/it/docs/Web/API/ServiceWorkerGlobalScope or https://jakearchibald.com/2016/service-worker-meeting-notes/).

IMHO global variables should be avoided whereever possible.

**Comment by ShinDarth** [comment]
I agree. We should definitely use a module loader. I would go for **Webpack** since it's one of the most populars and also the one used by default in [AngularCLI](https://github.com/angular/angular-cli), so it would make the project closer to the new Angular style.

**Comment by mk-pmb** [comment]
Have a look at [SystemJS](https://github.com/systemjs/systemjs). I was convinced when it transpiled [my all module formats hello world test](https://mk-pmb.github.io/tryjs/module_systems/systemjs/format-guessing/test.html) inside my Firefox and it just worked. For production you'd pre-transpile, of course.

**Comment by Ryuno-Ki** [comment]
I had trouble understanding SystemJS + JSPM, which prevents me from exploring Aurelia framework :-(

**Comment by mk-pmb** [comment]
I checked the [Aurelia Quick Start Guide](http://aurelia.io/hub.html#/doc/article/aurelia/framework/latest/quick-start/2) and to my delight it defaults to using system.js, so if this combination didn't work, it should work now, or you'll probably at least get support directly from Aurelia. The quick start guide also claimed they'll cover how to switch to production mode later.

**Comment by Ryuno-Ki** [comment]
Yeah, it was about a year or so back. Before their CLI. I assume, things are waaay better now. But we're getting off-topic here.

**Comment by mk-pmb** [comment]
> But we're getting off-topic here.

Really, do we? :-( I was hoping your idea was to replace Angular with Aurelia and WebComponents. :tada:

**Comment by Ryuno-Ki** [comment]
My idea is to support getting a decent test coverage here first in order to help zhukov finding people, who are actually able to *help* instead of just patching their forks.

Take a look at the contribution graphs. That's not healthy IMHO.
